### ISSUE TITLE
Add keyboard navigation

### ISSUE THREAD

**Issue by zepumph** [phetsims/molarity#37]
Like phetsims/ohms-law#49, we would like to add keyboard navigation to this sim. Tagging @emily-phet and @jessegreenberg so they are aware of this issue.

**Comment by pixelzoom** [comment]
Is there an objective that will be met by adding keyboard navigation to this sim specifically?

At last week's status meeting, I recommend to @jessegreenberg that we pick another sim.  While this sim is about to be published without the "Real Molecules" (3D) screen, that screen will be added in the future.  And keyboard navigation may or may not be straightforward for the 3D viewer, depending on what technology PhET decides to use.

**Comment by zepumph** [comment]
No sorry I wasn't aware of that. There is plenty of other work to be doing. Closing for now.

**Comment by pixelzoom** [comment]
Whoopsie.  I thought this issue was for molecule-polarity.  I did have a conversation with @jessegreenberg about molecule-polarity.  But molarity doesn't have 3D issues.   Do know that molarity was implemented before HSlider existed, and has its own odd slider requirements and implementation.  So the work done in phetsims/ohms-law#49 will not transfer.

**Comment by zepumph** [comment]
Oh yeah. @ariel-phet, @jessegreenberg and @emily-phet talked about that this morning in the design meeting for keyboard navigation. I think the consensus was that we would want to convert it to HSlider as part of adding keyboard navigation, and make sure you are in the loop as we go because it's your sim. This isn't high on my priority list, but was thinking about chipping away at it when accessibility time allowed.

**Comment by jessegreenberg** [comment]
Converting to HSlider would probably be best, but if there are features in VerticalSlider that make it difficult to convert, it could be a nice time to investigate https://github.com/phetsims/scenery/issues/607 which could let us mix in the behavior of an accessible slider to any scenery Node.

**Comment by pixelzoom** [comment]
> I think the consensus was that we would want to convert it to HSlider as part of adding keyboard navigation

That's going to be non-trivial.

**Comment by pixelzoom** [comment]
Conversion to HSlider will be handled in https://github.com/phetsims/molarity/issues/40.

**Comment by pixelzoom** [comment]
@jessegreenberg @zepumph Molarity has been converted to use HSlider, see #40.   Do you intend to add keyboard navigation in the 1.3 release branch, or only in master?  That will determine whether the HSlider conversion needs to be patched into 1.3.

**Comment by jessegreenberg** [comment]
Thanks or converting @pixelzoom.  Keyboard navigation will only be added to master, we won't be patching it into 1.3.

**Comment by zepumph** [comment]
Right now we are all working on OL, I will get to this a bit later on. Decreasing priority and unassigning for now.

**Comment by zepumph** [comment]
I updated a few small things in this issue. I made the shift modifier work for the hslider to achieve that maximum specificity with the slider.

@jessegreenberg it would be great to pair with you on combo box at some point. I don't think I know the accessibility side of life quite enough to tackle it efficiently/effectively on my own.

**Comment by jessegreenberg** [comment]
That sounds great @zepumph, lets do this sometime soon.

Here are my thoughts based on what we know so far:
- HTML select boxes won't work because it is impossible to style their content to be invisible in the DOM (we discovered this while working on build-an-atom).

- We will have to use some custom HTML with ARIA to represent select boxes. Here is a starting point: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_listbox_role
Ideally, we do NOT want to use aria-activedescendant because it's use will prevent scenery from placing default focus highlight around options. So it would be good to investigate a slightly alternative approach.

- Before instrumenting ComboBox, it would be good to make sure that PhET's supported AT support this markup.

**Comment by jessegreenberg** [comment]
@terracoda is going to take a look at best practices for accessible combo boxes. In a meeting  9/12/17, these resources were found:

https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/#combobox
http://demos.telerik.com/kendo-ui/combobox/keyboard-navigation

We are going to wait to implement until then. @terracoda please reassign to us when HTML examples are ready.

**Comment by pixelzoom** [comment]
@jessegreenberg Presumably the decisions and implementation required to add accessibility to ComboBox are not specific to Molarity.  Highly recommended to create a sun issue to capture those decisions and the related work.

**Comment by pixelzoom** [comment]
... and have @terracoda make recommendations in the sun issue.

**Comment by jessegreenberg** [comment]
Good idea, moving to sun.

**Comment by terracoda** [comment]
I will make my recommendations in the sun issue, https://github.com/phetsims/sun/issues/314, so closing this issue for now.
